sources/Algo/include/Algo.h

```diff
@@ -393,5 +398,300 @@ class ControllerInterfaceDefinitionAlgorithm : public NodeAlgorithm {
+     */
+    bool operator==(const MacroConnection& other) const;
+
+    /**
+     * @brief Non-equality operator
+     * @param other the macro connection to compare to
+     * @returns true if they are not equal, false if not
+     */
+    bool operator!=(const MacroConnection& other) const;
+
+    /**
+     * @brief Less than operator
+     * @param other the macro connection to compare to
+     * @returns true if inferior than other, false if not
+     */
+    bool operator<(const MacroConnection& other) const;
+
+    /**
+     * @brief Less or equal than operator
+     * @param other the macro connection to compare to
+     * @returns true if inferior or equal than other, false if not
+     */
+    bool operator<=(const MacroConnection& other) const;
+
+    /**
+     * @brief Greater than operator
+     * @param other the macro connection to compare to
+     * @returns true if superior than other, false if not
+     */
+    bool operator>(const MacroConnection& other) const;
+
+    /**
+     * @brief Greater or equal than operator
+     * @param other the macro connection to compare to
+     * @returns true if superior or equal than other, false if not
+     */
+    bool operator>=(const MacroConnection& other) const;
+
+    MacroId id;                    ///< Macro connector id
+    ElementType elementType;       ///< connected element type
+    ElementId connectedElementId;  ///< Element id connected throught the macro connection (can be node, line or tfo)
+  };
+
+  /**
+   * @brief Constructor
+   *
+   * @param idauto the dynamic model id
+   * @param libauto the library name
+   */
+  DynModelDefinition(const DynModelId& idauto, const std::string& libauto) : id(idauto), lib(libauto) {}
+
+  DynModelId id;                              ///< dynamic model id
+  std::string lib;                            ///< library name
+  std::set<MacroConnection> nodeConnections;  ///< set of macro connections for the dynamic model
+};
+
+/**
+ * @brief Definition of models
+ */
+struct DynModelDefinitions {
+  std::unordered_map<DynModelDefinition::DynModelId, DynModelDefinition> models;          ///< models by dynamic model id
+  std::unordered_set<DynModelDefinition::MacroConnection::MacroId> usedMacroConnections;  ///< list of macro connectors used for current set of models
+};
+
+/**
+ * @brief DynModel algorithm
+ */
+class DynModelAlgorithm : public NodeAlgorithm {
+ public:
+  /**
+   * @brief Constructor
+   *
+   * The constructor will pre-process some data in order relevant information to be retrieved with efficiency during main algorithm
+   *
+   * @param models the models to update
+   * @param manager the dynamic data base manager to use
+   */
+  DynModelAlgorithm(DynModelDefinitions& models, const inputs::DynamicDataBaseManager& manager);
+
+  /**
+   * @brief Perform the algorithm
+   *
+   * According to if the node is concerned in a macro connection, dispatch the node to update the macro connections in the models definition.
```

Comment:
I would propose "update the macro connections information in the models definition"

Reply:
OK